lib/puma/server.rb

```diff
@@ -589,18 +589,17 @@ def handle_request(req, lines)
 
             return :async
           end
-        rescue ThreadPool::ForceShutdown => e
-          @events.log "Detected force shutdown of a thread, returning 503"
-          @events.unknown_error self, e, "Rack app"
-
-          status = 503
-          headers = {}
-          res_body = ["Request was internally terminated early\n"]
-
         rescue Exception => e
           @events.unknown_error self, e, "Rack app", env
```

Comment:
Technically this means we pass `env` on shutdown errors, but that seemed like more of an oversight than an initial design. Happy to unwind if I'm wrong though.